raster-foundry/raster-foundry
# Anti-meridian search fixes, migration for fixing data_footprints

## Overview
- Support searching with multiple bounding boxes, so the frontend can make searches that cross the anti-meridian
- Add migration to split the data_footprint for any scenes that cross the anti-meridian from the following datasources: `Landsat 8, Sentinel 2, MYD09A1 (Modis), and ISERV (prod only)`

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

### Notes
- This migration does not fix scenes above 80 and below -80 because nobody cares about polar imagery right now and it's about 10x more complicated.
- This migration does not fix the import processes which generate invalid anti-meridian-crossing polygons, so it will need to be run a second time once we fix them.
- This does not fix anything which relies on a bounding box for the scene other than searches. Projects which contain a split scene will look like they cover the entire world (because they technically do). This only prevents those scenes, which nobody really cares too much about since they're just ocean, from showing up in searches where they shouldn't.

## Testing Instructions

* Verify that you can search using any number of bounding boxes
  - Test with 2 separate bounding boxes which are relatively far away from each other
  - Verify that the numbers add up as expected (eg: bbox 1 returns 10 scenes, bbox 2 returns 5 scenes, bbox 1 + 2 returns 15 scenes)
* Verify that after running the migration, scenes which cross the anti-meridian are now fixed and have multiple data_footprint polygons. 

example sql statement to fetch `data_footprint`'s from scenes that cross the positive anti-meridian:

```sql
select st_asgeojson(st_transform(data_footprint,4326)) from scenes where st_intersects(data_footprint, st_transform('SRID=4326;LINESTRING(180 -80, 180 80)'::geometry, 3857));
```
Also check the negative anti-meridian, `-180 -80, -180 80`, and the prime meridian `0 -80, 0 80` and validate them

Searching for scenes using the frontend should work, and you will be able to see the part of the bounding box which is on the same side of the anti-meridian as your map view is

Connects https://github.com/raster-foundry/raster-foundry/issues/3498
